tests/dnfserver/start.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+
+DIR=$(dirname "$(realpath "${BASH_SOURCE[0]}")")
+
+DEFAULT_IMAGE=quay.io/bcook/cachi2-nginx
+DEFAULT_PORT=8443
+
+cleanup() {
+    rm -r "$WORKDIR" || true
+    podman rmi -f quay.io/bcook/cachi2-nginx
+}
+
+WORKDIR=$(mktemp -d --tmpdir "cachi2-dnfserver.XXXXXX")
+trap cleanup EXIT
+
+setup() {
+    mkdir "$WORKDIR/certs"
+    echo -e "\n--- Starting dnfserver on http://localhost:${DNFSERVER_PORT:-8443} ---\n"
+}
+
+setup >&2
+
+podman run --rm --replace --name cachi2-dnfserver \
+    -p 8081:8081 \
```

Comment:
We don't need base HTTP for testing SSL right?

Reply:
it is used for aliveness check (without certs).

Reply:
> it is used for aliveness check (without certs).

If you want continuous check like this then that's what `--health-cmd` in podman is for, from outside it's not necessary to be exposed like this.

Reply:
I'm doing it [exactly the same way that the Pypi server is checked](https://github.com/brianwcook/cachi2/blob/3c562a5410ddd5f1043e7613b240bb5811682f7f/tests/integration/conftest.py#L69) in the equivalent test.

Reply:
Which is a good starting point. With the exception that the nginx service you're adding requires more configuration so I see value in some (pretty much any) automatic kind of connection checking based on the SSL certs.

Reply:
I've made the aliveness check use the ssl port with client certificates. If you want to use the podman health check, someone else can alter if for both dnfserver and pypi server later.

Reply:
>  If you want to use the podman health check, someone else can alter if for both dnfserver and pypi server later.

I'm okay with that, that was mostly just an idea I pitched, as long as we have some kind of an automated **pre-test** SSL check of the server, I'm good.